src/Builders/PromptBuilder.php

```diff
@@ -0,0 +1,1327 @@
+     *
+     * @return bool True if image generation is supported.
+     */
+    public function isSupportedForImageGeneration(): bool
+    {
+        return $this->isSupported(CapabilityEnum::imageGeneration());
+    }
+
+    /**
+     * Checks if the prompt is supported for text to speech conversion.
+     *
+     * @since n.e.x.t
+     *
+     * @return bool True if text to speech conversion is supported.
+     */
+    public function isSupportedForTextToSpeechConversion(): bool
+    {
+        return $this->isSupported(CapabilityEnum::textToSpeechConversion());
+    }
+
+    /**
+     * Checks if the prompt is supported for video generation.
+     *
+     * @since n.e.x.t
+     *
+     * @return bool True if video generation is supported.
+     */
+    public function isSupportedForVideoGeneration(): bool
+    {
+        return $this->isSupported(CapabilityEnum::videoGeneration());
+    }
+
+    /**
+     * Checks if the prompt is supported for speech generation.
+     *
+     * @since n.e.x.t
+     *
+     * @return bool True if speech generation is supported.
+     */
+    public function isSupportedForSpeechGeneration(): bool
+    {
+        return $this->isSupported(CapabilityEnum::speechGeneration());
+    }
+
+    /**
+     * Checks if the prompt is supported for music generation.
+     *
+     * @since n.e.x.t
+     *
+     * @return bool True if music generation is supported.
+     */
+    public function isSupportedForMusicGeneration(): bool
+    {
+        return $this->isSupported(CapabilityEnum::musicGeneration());
+    }
+
+    /**
+     * Checks if the prompt is supported for embedding generation.
+     *
+     * @since n.e.x.t
+     *
+     * @return bool True if embedding generation is supported.
+     */
+    public function isSupportedForEmbeddingGeneration(): bool
+    {
+        return $this->isSupported(CapabilityEnum::embeddingGeneration());
+    }
+
+    /**
+     * Generates a result from the prompt.
+     *
+     * This is the primary execution method that generates a result (containing
+     * potentially multiple candidates) based on the configured output modality.
+     *
+     * @since n.e.x.t
+     *
+     * @return GenerativeAiResult The generated result containing candidates.
+     * @throws InvalidArgumentException If the prompt or model validation fails.
+     * @throws RuntimeException If the model doesn't support the configured output modality.
+     */
+    public function generateResult(): GenerativeAiResult
+    {
+        $this->validateMessages();
+        $model = $this->getConfiguredModel();
+
+        // Get the configured output modalities
+        $outputModalities = $this->modelConfig->getOutputModalities();
+
+        // Default to text if no output modality is specified
+        if ($outputModalities === null || empty($outputModalities)) {
+            $outputModalities = [ModalityEnum::text()];
+        }
+
+        // Multi-modal output (multiple modalities) uses TextGenerationModelInterface
```

Comment:
Maybe clarify that eventually we want to have a more multimodal dedicated interface?
```suggestion
        /*
         * For now, multi-modal output (multiple modalities) uses TextGenerationModelInterface.
         * In the future, there will be an additional multimodal generation model interface that extenders can use.
         */
```

Reply:
Resolved in f366cdd